libvcx/src/lib.rs

```diff
@@ -659,4 +713,171 @@ mod tests {
+        let credential_handle2 = _exchange_credential(credential_data2, cred_def_handle, consumer_to_issuer, issuer_to_consumer, Some(consumer), Some(req2));
+
+        let proof_handle_verifier = _verifier_create_proof_and_send_request(&institution_did, &schema_id, &cred_def_id, verifier_to_consumer, Some(verifier), None);
+        _prover_select_credentials_and_send_proof(consumer_to_verifier, Some(consumer), None, None);
+        set_institution(Some(verifier));
+        proof::update_state(proof_handle_verifier, None, None).unwrap();
+        assert_eq!(proof::get_proof_state(proof_handle_verifier).unwrap(), ProofStateType::ProofValidated as u32);
+    }
+
+    #[test]
+    #[cfg(feature = "agency_pool_tests")]
+    fn test_two_creds_one_rev_reg_revoke_first() {
+        let _setup = SetupLibraryAgencyV2ZeroFees::init();
+        let institution_did = settings::get_config_value(settings::CONFIG_INSTITUTION_DID).unwrap(); // Issuer's did
+
+        let verifier = create_institution_config();
+        let consumer = create_consumer_config();
+        let (consumer_to_verifier, verifier_to_consumer) = ::connection::tests::create_connected_connections(Some(consumer), Some(verifier));
+        let (consumer_to_issuer, issuer_to_consumer) = ::connection::tests::create_connected_connections(Some(consumer), None);
+
+        let (schema_id, _schema_json, cred_def_id, _cred_def_json, cred_def_handle, rev_reg_id) = _create_address_schema();
+        let (address1, address2, city, state, zip) = attr_names();
+        let (req1, req2) = (String::from("request1"), String::from("request2"));
+        let credential_data1 = json!({address1.clone(): "123 Main St", address2.clone(): "Suite 3", city.clone(): "Draper", state.clone(): "UT", zip.clone(): "84000"}).to_string();
+        let credential_handle1 = _exchange_credential(credential_data1, cred_def_handle, consumer_to_issuer, issuer_to_consumer, Some(consumer), Some(req1));
+        let credential_data2 = json!({address1.clone(): "101 Tela Lane", address2.clone(): "Suite 1", city.clone(): "SLC", state.clone(): "WA", zip.clone(): "8721"}).to_string();
+        let credential_handle2 = _exchange_credential(credential_data2.clone(), cred_def_handle, consumer_to_issuer, issuer_to_consumer, Some(consumer), Some(req2));
+
+        revoke_credential(credential_handle1, rev_reg_id);
+
+        let proof_handle_verifier = _verifier_create_proof_and_send_request(&institution_did, &schema_id, &cred_def_id, verifier_to_consumer, Some(verifier), None);
+        _prover_select_credentials_and_send_proof(consumer_to_verifier, Some(consumer), None, Some(&credential_data2));
+        set_institution(Some(verifier));
+        proof::update_state(proof_handle_verifier, None, None).unwrap();
+        assert_eq!(proof::get_proof_state(proof_handle_verifier).unwrap(), ProofStateType::ProofValidated as u32);
```

Comment:
perhaps we could check verification for both issued credentials? That would apply to other tests as well